src/Microsoft.IdentityModel.Tokens/Validation/ValidationParameters.cs

```diff
@@ -228,6 +228,11 @@ public virtual ClaimsIdentity CreateClaimsIdentity(SecurityToken securityToken,
         /// </summary>
         public string DebugId { get; set; }
 
+        /// <summary>
+        /// Gets the <see cref="SecurityKey"/> representing the ephemeral public key used for ECDH-ES decryption.
+        /// </summary>
+        public SecurityKey EcdhEsDecryptionKey { get; set; }
```

Comment:
This is sensitive information, could be the private key, do we need this?

Reply:
I was reading to familiarise myself more with ECDH-ES as we currently support it and it appears to me that this is a public key that is also ephemeral, so I would not imagine it is too sensitive.

On the current implementation, the key is set in TVP.DecryptionKey or very similar. Since we are doing away with that field in favour of the List one, I think it is safest to allow the user to set this in a field specific for this use rather than iterating the list of keys and seeing if any is of this type.

Reply:
This is not printed anywhere.